src/Formatter/NMEAFormatter.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2023 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "Units/Units.hpp"
+#include "NMEAFormatter.hpp"
+
+void
+FormatGPRMC(char *buffer, size_t size, const NMEAInfo &info) noexcept
+{
+  const GeoPoint location = info.location_available
+    ? info.location
+    : GeoPoint::Zero();
```

Comment:
`GeoPoint::Zero()` is a valid location that exists on earth. I don't like using "magic" values like that. Can't we just keep the columns empty if we have no location, instead of printing an arbitrary bogus location?

Reply:
While I agree on your remark of "magic" values, I don't know how devices react when the latitude and longitude fields are left empty, together with the N/S/W/E identifiers. Most specifications of the NMEA0183 GPRMC sentence that I found don't mention that case, while a few just set it to 0000.0000. This is of course a valid location, but devices should discard the sentence based on field #2, which sets the position status to valid or invalid.